### ISSUE TITLE
Better support for nested collection properties for models annotated with @ModelAttribute

### ISSUE THREAD

**Issue by shirishp** [springfox/springfox#1134]
This might be duplicate / related to #969 
But I just wanted to make sure. Please let me know if this is different.
I am using

```groovy
    compile 'io.springfox:springfox-swagger2:2.2.2'
    compile 'io.springfox:springfox-swagger-ui:2.2.2'
```
I have a class which I am using as `@ModelAttribute`. This class has an attribute of type `List<String>`, with getter and setter. But in the generated schema, I can see it referred as:

```javascript
{
   "name": "passengerNumbers",
    "in": "query",
    "required": false,
    "type": "ref"
}
```
While, spring is able to populate the List if I pass 2 values for the field from Postman, swagger-ui doesn't show this as List. It shows it's datatype as `undefined`.

Is it known issue or am I missing something? Please let me know if you need any more information to debug this.

Thanks for the great project. :)

**Comment by shirishp** [comment]
Sorry to bother again, but did you get chance to have a look at this?

Also, to add more information, If I pass another parameter `@RequestParam("somethingElse) List<String> somethingElse` along with `@ModelAttribute`, Swagger schema for `somethingElse` is generated properly, but not the field from `@ModelAttribute` class.

**Comment by dilipkrish** [comment]
@shirishp support for `@ModelAttribute` Is very basic. The reason being these are not really json services in the traditional sense. Secondly array/matrix parameters are don't translate very well across languages in the way they are represented. I.e what would the parameter name be somethingElse[] vs. repeated somethingElse? How do List of complex types look etc. 

Having said that I would chalk this up as a feature request. Would love to get a pr if possible.

**Comment by shirishp** [comment]
I would love to work on PR. Would it be possible for you to guide me to the section of the code?

**Comment by dilipkrish** [comment]
Absolutely! Much appreciated :bow: :metal: 

I'll create a set of steps to guide you perhaps.

**Comment by dilipkrish** [comment]
This [should get you started](https://github.com/springfox/springfox/issues/182#issuecomment-171133102) getting familiar with the architecture. In your case you want to look at [ModelAttributeParameterExpander](https://github.com/springfox/springfox/blob/master/springfox-spring-web/src/main/java/springfox/documentation/spring/web/readers/parameter/ModelAttributeParameterExpander.java) and [ExpandedParameterBuilder](https://github.com/springfox/springfox/blob/master/springfox-spring-web/src/main/java/springfox/documentation/spring/web/readers/parameter/ExpandedParameterBuilder.java).

**Comment by anand4sn** [comment]
Hi All,

  I'm using @ModelAttribute for Get method, the Model object has mutiple properties like, List<String> accounts, String language,String eventType ,i have used @ApiModelProperty for all the model properties. My problem is swagger documentation is not showing  the descriptions.i attached screenshot here. could you please help on this.
![image](https://cloud.githubusercontent.com/assets/2691445/12327046/438c34fa-bafa-11e5-8f98-40d67ed30e01.png)

**Comment by dilipkrish** [comment]
@anand4sn just a quick note.. even if this issue is fixed, swagger-ui will not be able to render that or help with editing the list field in the UI

**Comment by anand4sn** [comment]
@dilipkrish   Many Thanks

**Comment by dilipkrish** [comment]
This is fixed via #1317
